### PR TITLE
[AUTO-1198] Fix cypress reporter & add jest tests

### PR DESCRIPTION
**Pull request by josiah-poss**
https://vaulthealth.atlassian.net/browse/AUTO-1198

## Summary

- Fix Cypress Reporter by updating the `suiteId === platformSuiteIds["MONOREPO"]` to double equals 
- Create an Enum for Suites (we will have a lot more in the future) 
- Create new function to handle figuring out the suiteName
    - Add Default case, for when neither Suites are selected 

- Add JEST testing to validate the new method showing that both integers & strings work for the method

## Screenshots 
Screen Shot 2022-08-02 at 12 18 17 PM

## Outstanding issues
- [x]  I probably added more packages than needed, will try to reduce these additions
- [x] Rename the function that decides the name
- [ ] Add README around running JEST tests locally

https://www.npmjs.com/package/ts-jest

### CODE REVIEW

File: src/lib/cypress-testrail-reporter.ts [hunk 1]
```diff
@@ -160,12 +150,31 @@ export class CypressTestRailReporter extends reporters.Spec {
...
   }
 }
+
+/**
+ * Decide which suite name should be assigned 
+ * @param suiteId current Suite running
+ * @param runName 
+ * @returns 
+ */
+export function evaluateActiveSuite(suiteId, runName?): string {
```
Comment (josiah-poss) [inline src/lib/cypress-testrail-reporter.ts:169]: Trying to think of a better name for this
```
generateSuiteName()
evaluateAndGenerateSuiteName()
getSuiteName()
...
```

File: package.json [hunk 2]
```diff
@@ -34,11 +35,15 @@
...
     "@types/chai": "^3.5.0",
+    "@types/jest": "^28.1.6",
     "@types/mocha": "^8.2.2",
     "@types/node": "^14.14.41",
     "chai": "^3.5.0",
+    "jest": "^28.1.3",
     "rimraf": "^2.7.1",
-    "typescript": "^4.2.4"
+    "ts-jest": "^28.0.7",
+    "ts-node": "^10.9.1",
```
Comment (josiah-poss) [inline package.json:45]: `ts-node` probably isn't needed. 

We can also probably just import `ts-jest`

**Comment by josiah-poss** [comment]
@zhivou @Seminlee94 any concerns that adding JEST will cause issues with the reporter? 

Also, the main fix for this was just not use `===` to choose SuiteName

File: package.json [hunk 3]
```diff
@@ -34,10 +35,13 @@
   },
   "devDependencies": {
     "@types/chai": "^3.5.0",
+    "@types/jest": "^28.1.6",
```
Comment (Amir-Gudic) [inline package.json:38]: do we need this for TS type support? many types can be used from jest alone as I recall?

**Review by Amir-Gudic — APPROVED** [review]
Awesome!
Thanks, @josiah-poss !
Hoping for more of this in the future from all of us :)

**Comment by Seminlee94** [comment]
everything looks good, but @zhivou I kinda forgot about dist and src, but don't we need to update dist folder as well so it reflects whatever @josiah-poss changed in src folder?

File: package.json (hunk 3 above)
Comment (josiah-poss) [reply package.json:38]: Turns out we don't need this! I will update it in the next push

**Pull request closed without merging**